client/src/modules/dashboard/Events/components/EventForm.tsx

```diff
@@ -108,6 +136,88 @@ const EventForm: React.FC<EventFormProps> = (props) => {
           </FormControl>
         )}
 
+        <FormControl id="first-name" isRequired>
+          <Box display="flex" alignItems="end" m="1">
+            <FormLabel> Sponsers</FormLabel>
+            <Spacer />
+            <Button
+              onClick={() => {
+                append({
+                  type: 'FOOD',
+                  id: 0,
+                });
+              }}
+            >
+              Add
+            </Button>
+          </Box>
+          {sponsors.map((sponsor, index) => {
+            return (
+              <Flex key={sponsor.key} borderWidth="1px" p="5" mb="5">
+                <Box display="flex" flexGrow={1}>
+                  <FormControl m="1">
+                    <FormLabel>Sponsor Type</FormLabel>
+                    <Select
+                      defaultValue={getValues(`sponsors.${index}.type`)}
+                      {...register(`sponsors.${index}.type` as const, {
+                        required: true,
+                      })}
+                    >
+                      <option value="FOOD">Food</option>
+                      <option value="VENUE">Venue</option>
+                      <option value="OTHER">Other</option>
+                    </Select>
+                  </FormControl>
+
+                  <FormControl m="1">
+                    <FormLabel> Sponsor Name</FormLabel>
+                    {loadingSponsors ? (
+                      <h5>loading sponsors</h5>
+                    ) : errorSponsor || !sponsorData ? (
+                      <h5> Error loading sponsors</h5>
+                    ) : (
+                      <Select
+                        defaultValue={getValues(`sponsors.${index}.id`)}
+                        {...register(`sponsors.${index}.id` as const, {
+                          required: true,
+                        })}
+                      >
```

Comment:
I've encountered a problem when running this locally.  I haven't been able to pin down exactly how to reproduce this, but sometimes it tries to send 0 as a sponsor id, which the DB rejects.

Also, this is a bit odd:
![SponsorsShifting](https://user-images.githubusercontent.com/15801806/139917548-90da9e3a-8c5e-41d1-97e6-6fe77f95e3a9.gif)
 
That's on /dashboard/events/[id]/edit

Finally, if the updates are successfully submitted, you have to reload the page before the sponsor changes appear in the form. Other fields seem to be fine.

Reply:
>Finally, if the updates are successfully submitted, you have to reload the page before the sponsor changes appear in the form. Other fields seem to be fine.

@ojeytonwilliams , How can we overcome this problem?

Reply:
>If you've got an idea what's going wrong, awesome. If not, it might help to simplify the form and not filter for already selected sponsor names. We can always revisit that in another PR if the core behaviour is working.

@ojeytonwilliams , I've removed the code for filtering out the selected options.

>but sometimes it tries to send 0 as a sponsor id, which the DB rejects.

I too can't reproduce it. but my best guess is that, when we click on the Add button, we are by default setting the id to 0. and if user Click Save button then, id might be sent as zero. After removing the filter code , I'm not able to reproduce it my local.

Reply:
@Ravichandra-C I have not had chance to look into the re-fetching issue, but I'll let you know when I have something!